src/app/content/highlights/components/Card.tsx

```diff
@@ -83,16 +70,30 @@ const Card = (props: Props) => {
     }
   }, [props.highlight, annotation]);
 
-  const {page, book} = props;
+  React.useEffect(() => {
+    if (!annotation && !props.isFocused) {
+      props.onHeightChange({ current: null });
+    }
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [annotation, props.isFocused]);
+
+  const location = React.useMemo(() => {
+    return props.page && getHighlightLocationFilterForPage(locationFilters, props.page);
+  }, [locationFilters, props.page]);
+
+  const locationFilterId = location && stripIdVersion(location.id);
 
-  if (!props.highlight.range || !page || !book) {
+  const { page, book } = props;
+  if (!props.highlight.range || !page || !book || !locationFilterId || (!props.isFocused && !annotation)) {
     return null;
   }
 
-  const location = getHighlightLocationFilterForPage(locationFilters, page);
-  if (!location) { return null; }
-
-  const locationFilterId = stripIdVersion(location.id);
+  const handleClickOnCard = async(e: React.MouseEvent) => {
+    if (!props.isFocused && (!hasUnsavedHighlight || await showConfirmation())) {
+      e.preventDefault();
```

Comment:
It was added here: https://github.com/openstax/rex-web/commit/60aefe2a5c46b31630fe8cefbdbc83c33464b4e5but it might not be required anymore due to change how resetting top offset for card wrapper works now